src/main/java/http_server/Settings.java

```diff
@@ -0,0 +1,54 @@
+package http_server;
+
+import java.util.HashMap;
+
+public class Settings {
+    private static HashMap<String, String> settings = new HashMap<>();
+
+    public static int getPort() {
+        return Integer.parseInt(settings.get("port"));
+    }
+
+    public static String getDirectory() {
+        return settings.get("directory");
+    }
+
+    public static boolean validateSettings(String[] args) {
+        initializeSettings();
+
+        for (int i = 0; i < args.length; i++) {
+            if (i + 1 < args.length) {
+                if (args[i].equals("-p")) {
```

Comment:
There are a lot of conditionals in here. It would be easy to extract a couple of them...